client/v2/indexer/lookupAssetTransactions.go

```diff
@@ -0,0 +1,100 @@
+package indexer
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/client/v2/common"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+)
+
+type LookupAssetTransactions struct {
+	c     *Client
+	index uint64
+	p     models.LookupAssetTransactionsParams
+}
+
+func (s *LookupAssetTransactions) NextToken(nextToken string) *LookupAssetTransactions {
+	s.p.NextToken = nextToken
+	return s
+}
+
+func (s *LookupAssetTransactions) NotePrefix(prefix []byte) *LookupAssetTransactions {
+	s.p.NotePrefix = base64.StdEncoding.EncodeToString(prefix)
+	return s
+}
+
+func (s *LookupAssetTransactions) TxType(txtype string) *LookupAssetTransactions {
+	s.p.TxType = txtype
+	return s
+}
+
+func (s *LookupAssetTransactions) SigType(sigtype string) *LookupAssetTransactions {
+	s.p.SigType = sigtype
+	return s
+}
+
+func (s *LookupAssetTransactions) TXID(txid string) *LookupAssetTransactions {
+	s.p.TxId = txid
+	return s
+}
+
+func (s *LookupAssetTransactions) Round(rnd uint64) *LookupAssetTransactions {
+	s.p.Round = rnd
+	return s
+}
+
+func (s *LookupAssetTransactions) MinRound(min uint64) *LookupAssetTransactions {
+	s.p.MinRound = min
+	return s
+}
+
+func (s *LookupAssetTransactions) MaxRound(max uint64) *LookupAssetTransactions {
+	s.p.MaxRound = max
+	return s
+}
+
+func (s *LookupAssetTransactions) Address(address string) *LookupAssetTransactions {
```

Comment:
nit: this could be our address type, that would ensure that the address was valid before making the request.